File: Chapter02/PostgreSQL/Maven/DeclarativeJPADatabase/webapp/src/main/java/com/classicmodels/repository/ClassicModelsRepositoryImpl.java
package com.classicmodels.repository;

import com.classicmodels.pojo.EmployeeNoCntr;
import java.util.List;
import static jooq.generated.tables.Employee.EMPLOYEE;
import org.jooq.DSLContext;
import static org.jooq.impl.DSL.firstValue;
import org.springframework.stereotype.Repository;

@Repository
public class ClassicModelsRepositoryImpl implements ClassicModelsRepository {

    private final DSLContext ctx;

    public ClassicModelsRepositoryImpl(DSLContext ctx) {
        this.ctx = ctx;               
    }

    @Override
    public List<EmployeeNoCntr> findEmployeesAndLeastSalary() {        
        
        ctx.configuration().settings()
                 .withRenderNameCase​(org.jooq.conf.RenderNameCase.LOWER);
        
        List<EmployeeNoCntr> result = ctx.select(EMPLOYEE.FIRST_NAME,
                EMPLOYEE.LAST_NAME,
                EMPLOYEE.SALARY,
                firstValue(EMPLOYEE.FIRST_NAME)
                        .over().orderBy(EMPLOYEE.SALARY).as("least_salary"))
                .from(EMPLOYEE)
                .fetchInto(EmployeeNoCntr.class);

        return result;
    }
}

File: Chapter02/Kotlin/Oracle/Maven/DeclarativeJPADatabase/webapp/src/main/kotlin/com/classicmodels/repository/ClassicModelsRepositoryImpl.kt
package com.classicmodels.repository

import com.classicmodels.pojo.EmployeeNoCntr
import jooq.generated.tables.references.*
import org.jooq.DSLContext
import org.springframework.stereotype.Repository
import org.jooq.conf.RenderNameCase
import org.jooq.impl.DSL.firstValue

@Repository
class ClassicModelsRepositoryImpl(private val ctx: DSLContext) : ClassicModelsRepository {

    override fun findEmployeesAndLeastSalary(): MutableList<EmployeeNoCntr> {

       (ctx.configuration().settings()::setRenderNameCase)(RenderNameCase.UPPER) // this is default anyway

       return ctx.select(EMPLOYEE.FIRST_NAME, EMPLOYEE.LAST_NAME, EMPLOYEE.SALARY,
                firstValue(EMPLOYEE.FIRST_NAME)
                        .over().orderBy(EMPLOYEE.SALARY).`as`("least_salary"))
                .from(EMPLOYEE)
                .fetchInto(EmployeeNoCntr::class.java)
    }

}